uptane-standard.md

```diff
@@ -110,26 +110,116 @@ TODO
 
 ## Roles
 
+In order to read and write metadata, it helps to understand that all metadata on
+a repository is produced by one of four basic roles: root, targets, snapshot,
+and timestamp. The relationships between these roles are illustrated in Figure
+2a, whereas Table 2a summarizes their respective responsibilities. To better
+understand the design behind these roles, the reader should read the
+[CCS’10 paper](https://ssl.engineering.nyu.edu/papers/samuel_tuf_ccs_2010.pdf)
+on The Update Framework (TUF), a precursor to Uptane.
+
+Figure 2a: The four basic (root, timestamp, snapshot, and targets) roles used
+in Uptane, along with the corresponding metadata and images.
+
+Table 2a: The four basic roles that Uptane uses to add signed metadata to a
+repository.
+
 ### The Root Role
 
+The root role serves as the certificate authority. It distributes and revokes
+the public keys used to verify metadata produced by each of the four basic roles
+(including itself).
+
 ### The Targets Role and Delegations
 
+The targets role provides metadata, such as hashes and file sizes, about images.
+
+Instead of signing metadata itself, the targets role MAY delegate the
+responsibility of signing this metadata to other, custom-made roles. For
+example, in Figure 2a, the targets role has delegated all images that match the
+filename pattern “A.*” to the A1 role, and all images that match the filename
+patterns “B.*” and “C.*” to the BC role. In turn, the A1 role delegates a subset
+of its images (in this case, only the “A.img”) to the A2 role. A delegation
+binds the public keys used by a delegatee to a subset of the images these keys
+are trusted to sign. This means that the targets role distributes and revokes
+the public keys for the A1 and BC roles, whereas the A1 role does the same for
+the A2 role.
+
+There are two types of delegations: prioritized and / or terminating
+delegations. Furthermore, a delegation may require multiple roles, as explained
+below.
+
 #### Prioritized Delegations
 
+Sometimes an image may be delegated to more than one role. For example, the
```

Comment:
Section should begin with a succinct definition of prioritized delegations, rather than an explanation of the rationale for existing